include/seqan3/range/views/minimiser.hpp

```diff
@@ -0,0 +1,550 @@
+// Copyright (c) 2006-2020, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2020, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \author Mitra Darvish <mitra.darvish AT fu-berlin.de>
+ * \brief Provides seqan3::views::minimiser.
+ */
+
+#pragma once
+
+#include <deque>
+
+#include <seqan3/range/hash.hpp>
+
+namespace seqan3::detail
+{
+// ---------------------------------------------------------------------------------------------------------------------
+// minimiser class
+// ---------------------------------------------------------------------------------------------------------------------
+
+/*!\brief The type returned by seqan3::views::minimiser.
+ * \tparam urng_t The type of the underlying ranges, must model std::forward_range, the reference type must model
+ *                std::size_t. The typical use case is that the reference type is the result of seqan3::kmer_hash.
+ * \implements std::ranges::view
+ * \implements std::ranges::random_access_range
+ * \implements std::ranges::sized_range
+ * \ingroup views
+ *
+ * \details
+ *
+ * Note that most members of this class are generated by ranges::view_interface which is not yet documented here.
+ */
+template <std::ranges::view urng_t>
+class minimiser : public std::ranges::view_interface<minimiser<urng_t>>
+{
+private:
+    static_assert(std::ranges::forward_range<urng_t const>, "The minimiser only works on forward_ranges");
+
+    //!\brief The underlying range.
+    urng_t urange;
+
+    //!\brief The number of elements in one window.
+    uint32_t w_elems;
+
+    template <typename rng_t>
+    class window_iterator;
+
+public:
+    /*!\name Constructors, destructor and assignment
+     * \{
+     */
+    minimiser()                                  = default; //!< Defaulted.
+    minimiser(minimiser const & rhs)             = default; //!< Defaulted.
+    minimiser(minimiser && rhs)                  = default; //!< Defaulted.
+    minimiser & operator=(minimiser const & rhs) = default; //!< Defaulted.
+    minimiser & operator=(minimiser && rhs)      = default; //!< Defaulted.
+    ~minimiser()                                 = default; //!< Defaulted.
+
+    //!\brief Construct from a view and a given number of elements in one window (w_elems).
+    minimiser(urng_t urange_, uint32_t const & w_) : urange{std::move(urange_)}, w_elems{w_}{}
+
+    //!\brief Construct from a non-view that can be view-wrapped and a given number of elements in one window (w_elems).
+    template <typename rng_t>
+    //!\cond
+     requires !std::same_as<seqan3::remove_cvref_t<rng_t>, minimiser> &&
```

Comment:
Do we exclude minimiser for technical reasons or because there is no application yet?
Technically a minimiser returns a range of numbers and I could use a minimiser on this range again...